File: grind75/Jan15/252.go
func canAttendMeetings(intervals [][]int) bool {
    if len(intervals) <= 1 { return true }
    sort.Slice(intervals, func(i,j int) bool {
        return intervals[i][0] < intervals[j][0]
    })
    i, j := 0, 1
    for j < len(intervals) {
        if intervals[i][1] > intervals[j][0] {
            return false
        }
        i++
        j++
    }
    return true
}

File: grind75/Jan15/252.py
class Solution:
    def canAttendMeetings(self, intervals: List[List[int]]) 
-> bool:
        if len(intervals) < 2:
            return True
        intervals.sort(key = lambda x: x[0])
        for i in range(1, len(intervals)):
            if intervals[i][0] < intervals[i-1][1]:
                return False
        return True
